Branch-SDK/Branch-SDK/ContentDiscoverer.m

```diff
@@ -0,0 +1,287 @@
+                }
+                
+                [[BNCPreferenceHelper preferenceHelper]saveBranchAnalyticsData:contentEventObj];
+            }
+        }
+    }
+}
+
+
+
+- (void)discoverViewContents:(UIView *)rootView contentData:(NSMutableArray *)contentDataArray contentKeys:(NSMutableArray *)contentKeysArray clearText:(BOOL)isClearText ID:(NSString *)viewId {
+    if ([rootView isKindOfClass:UITableView.class]) {
+        NSArray * cells = ((UITableView *)rootView).visibleCells;
+        int cellCnt = -1;
+        for (UIView* cell in cells) {
+            cellCnt++;
+            NSString *format;
+            if (viewId.length > 0 ) {
+                format = @"-%d";
+            } else {
+                format = @"%d";
+            }
+            NSString *cellViewId = [viewId stringByAppendingFormat:format ,cellCnt];
+            [self discoverViewContents:cell contentData:contentDataArray contentKeys:contentKeysArray clearText:isClearText ID:cellViewId];
+        }
+    } else {
+        if ([rootView respondsToSelector:@selector(text)]) {
+            NSString *contentData;
+            contentData = [self getContentText:rootView];
+            if (contentData != nil) {
+                NSString * viewFriendlyName = [NSString stringWithFormat:@"%@:%@",[rootView class],viewId];
+                [contentKeysArray addObject:viewFriendlyName];
+                if (contentDataArray != nil) {
+                    [self addFormatedContentData:contentDataArray withText:contentData clearText:isClearText];
+                }
+            }
+        }
+        NSArray *subViews = [rootView subviews];
+        if (subViews.count > 0) {
+            int childCount = -1;
+            for (UIView* view in subViews ) {
+                childCount++;
+                NSString *subViewId = [viewId stringByAppendingFormat:@"-%d" ,childCount];
+                [self discoverViewContents:view contentData:contentDataArray contentKeys:contentKeysArray clearText:isClearText ID:subViewId];
+            }
+        }
+    }
+}
+
+
+- (void)discoverFilteredViewContents:(NSMutableArray *)contentDataArray contentKeys:(NSMutableArray *)contentKeysArray clearText:(BOOL)isClearText {
+    for (NSString * contentKey in contentKeysArray) {
+        NSString *contentData = [self getViewText:contentKey forController:lastViewController];
+        if (contentData == nil) {
+            contentData = @"";
+        }
+        if (contentDataArray != nil) {
+            [self addFormatedContentData:contentDataArray withText:contentData clearText:isClearText];
+        }
+    }
+}
+
+
+- (NSString *)getViewText:(NSString *)viewId forController:(UIViewController *)viewController {
+    NSString *viewTxt = @"";
+    if (viewController != nil) {
+        UIView * rootView = [viewController view];
+        if ([viewId containsString:@":"]) {
+            NSArray * viewIDsplitArray = [viewId componentsSeparatedByString:@":"];
```

Comment:
This method returns the string if the separator is not present. If so, you don't need to check if it contains `:` first.